portal-kernel/src/com/liferay/portal/kernel/portlet/LiferayStateAwareResponse.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.kernel.portlet;
+
+import aQute.bnd.annotation.ProviderType;
+
+import java.util.List;
+
+import javax.portlet.Event;
+import javax.portlet.StateAwareResponse;
+
+/**
+ * @author Neil Griffin
+ */
+@ProviderType
+public interface LiferayStateAwareResponse
```

Comment:
@shuyangzhou Do you want me to remove LifearyStateAwareResponse from the interface hierarchy and add these common methods to both the LiferayActionResponse and LiferayEventResponse interfaces instead?

Reply:
@shuyangzhou I actually reference `if (actionResponse instanceof LiferayStateAwareResponse) {` in PortalImpl.java so I would like to keep LiferayStateAwareResponse if you would permit.

Reply:
@ngriffin7a it is ok to keep that one, as it really does something (holding those methods, and other code uses those method through this interface, right?). I just don't want to see some interfaces that have absolutely no usage.

Reply:
@shuyangzhou That is correct. OK I will keep LiferayStateAwareResponse but will remove LiferayClientDataRequest LiferayMimeResponse since they don't have any method signatures and are not referenced by the code.